File: port/Player.py
class Player:
    """
    If you want to create a new AI, you need to implement this interface.
    """

    def make_action(self, map, team_color, is_the_beginning_of_turn, is_the_first_turn_of_game):
        """
        Function to create the AI's action.
        :param map: Current MAP situation
        :param team_color: Team to operate
        :param is_the_beginning_of_turn: Whether it's the beginning of the turn
        :param is_the_first_turn_of_game: Whether it's the first turn of the game
        :return: AI's action (for 1 unit)
        """
        pass

    def get_name(self):
        """
        Function to return the name in one line.
        :return: Name of the AI
        """
        pass

    def show_parameters(self):
        """
        Function to return parameter information, which can include line breaks.
        :return: Parameter information
        """
        pass

File: Player.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace SimpleWars
{
    // 新規にAIを作成したい場合は，このインターフェースを
    // 実装する必要がある
    interface Player {
        //AIの行動作成関数
        /// <summary>
        /// AIの行動を作成する関数
        /// </summary>
        /// <param name="map">現在のMAP状況</param>
        /// <param name="teamColor">操作するチーム</param>
        /// <param name="turnStart">ターンの開始直後であるか</param>
        /// <param name="gameStart">ゲームの開始直後であるか</param>
        /// <returns>AIの行動（1ユニット分）</returns>
		Action makeAction(Map map, int teamColor, bool isTheBeggingOfTurn, bool isTheFirstTurnOfGame);

        // １行で，名前を返す関数
        string getName();

        // 改行を含んでも良い，パラメータ等の情報を返す関数
        string showParameters();

    }
}
